### PR TITLE
YOLO Family: Darknet backbones

### PR DESCRIPTION
**Pull request by anivegesana**
# Description

This pull request adds the backbones of the YOLOv3 and v4 families (Darknet53, TinyYOLOv3, CSPDarknet53, and TinyYOLOv4).

To train the CSPDarknet53 image classification backbone on ImageNet, use the following command in the official directory:
```
python3 -m official.vision.beta.projects.yolo.train --mode=train_and_eval --experiment=darknet_classification --model_dir=training_dir --config_file=official/vision/beta/projects/yolo/configs/experiments/darknet53.yaml
```

## Type of change

For a new feature or function, please create an issue first to discuss it
with us before submitting a pull request.

Note: Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation update
- [ ] TensorFlow 2 migration
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] A new research paper code implementation
- [ ] Other (Specify)

## Tests

Tests are provided for each of the building blocks. The tests provide checking to see if the outputs or gradients of the layers go to `None` anywhere in the layer.

**Test Configuration**: Currently, we use `unittest` in the Python3 standard library in order to verify the correctness of our code.

## Checklist

- [x] I have signed the [Contributor License Agreement](https://github.com/tensorflow/models/wiki/Contributor-License-Agreements).
- [x] I have read [guidelines for pull request](https://github.com/tensorflow/models/wiki/Submitting-a-pull-request).
- [x] My code follows the [coding guidelines](https://github.com/tensorflow/models/wiki/Coding-guidelines).
- [x] I have performed a self [code review](https://github.com/tensorflow/models/wiki/Code-review) of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

**Comment by anivegesana** [comment]
@googlebot I consent.

**Comment by The-Indian-Chinna** [comment]
@googlebot I consent.

**Review by jaeyounkim — APPROVED** [review]

**Review by arashwan — APPROVED** [review]
